### ISSUE TITLE
NPE / Spring Security plugin requires change in bean resolution

### ISSUE THREAD

**Issue by jmlittle** [graceframework/grace-framework#405]
### Describe the bug

As noted here, grails 5.3.x series had a fix to address support for spring security core to avoid NPE. https://github.com/grails/grails-core/issues/13250 

### Expected behavior

spring security is configured on startup

### Actual behavior

java.lang.NullPointerException: Cannot invoke method resolveType() on null object

### How to Reproduce?

On a new project, you can make a symlink from apps to grails-app and follow normal spring security core inclusion to a project, using s2-quickstart and writing to application.groovy. On running an app with the steps taken, an NPE occurs.

### Grace version (ie. output of `grace -v`)

2022.2.6

**Comment by jmlittle** [comment]
Relative to your variant, an update to grace-plugin-api is likely warranted:

```
diff --git a/grace-plugin-api/src/main/groovy/org/grails/plugins/DefaultGrailsPlugin.java b/grace-plugin-api/src/main/groovy/org/grails/plugins/DefaultGrailsPlugin.java
index 57be33facf..9f99a27eae 100644
--- a/grace-plugin-api/src/main/groovy/org/grails/plugins/DefaultGrailsPlugin.java
+++ b/grace-plugin-api/src/main/groovy/org/grails/plugins/DefaultGrailsPlugin.java
@@ -575,7 +575,7 @@ public class DefaultGrailsPlugin extends AbstractGrailsPlugin implements ParentA
                 bb.setBeanBuildResource(new DescriptiveResource(this.plugin.getClass().getName()));
                 bb.setBinding(b);
                 c.setDelegate(bb);
-                c.setResolveStrategy(Closure.DELEGATE_FIRST);
+                c.setResolveStrategy(Closure.OWNER_FIRST);
                 bb.invokeMethod("beans", new Object[] { c });
             }
         }
```

**Comment by jmlittle** [comment]
I did make this change and rebuilt a copy of grace, but there error persists. Not sure if grails-plugin-api to alter the DELEGATE_FIRST or somewhere else.

**Comment by rainboyan** [comment]
@jmlittle If you want to try and build it yourself, you can check out the related issue [GF-269](https://github.com/graceframework/grace-framework/issues/269).

But I don't think it is the best way to fix, because it will cause DynamicPlugin failure to work. I will spend some time on this issue after the new feature Application Templates is finished.

**Comment by rainboyan** [comment]
Closed via 4de88c59
